src/test/java/com/iota/iri/conf/ConfigFactoryTest.java

```diff
@@ -0,0 +1,170 @@
+     * <code>testnet: false</code> as method parameter for {@link ConfigFactory#createFromFile(File, boolean)}.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTestnetFalseAndFalse() throws IOException {
+        // lets assume in our configFile is TESTNET=false
+        File configFile = createTestnetConfigFile("false");
+
+        // but the parameter is set to testnet=true
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, false);
+        assertTrue("Expected iotaConfig as instance of MainnetConfig.", iotaConfig instanceof MainnetConfig);
+        assertFalse("Expected iotaConfig as Mainnet.", iotaConfig.isTestnet());
+    }
+
+    /**
+     * Test if leading and trailing spaces are trimmed from string in properties file.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithTrailingSpaces() throws IOException {
+        File configFile = createTestnetConfigFile("true");
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, true);
+        String expected = "NPCRMHDOMU9QHFFBKFCWFHFJNNQDRNDOGVPEVDVGWKHFUFEXLWJBHXDJFKQGYFRDZBQIFDSJMUCCQVICI";
+        assertEquals("Expected that leading and trailing spaces were trimmed.", expected, iotaConfig.getCoordinator());
+    }
+
+    /**
+     * Test if trailing spaces are correctly trimmed from integer.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithInteger() throws IOException {
+        File configFile = createTestnetConfigFile("true");
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, true);
+        assertEquals("Expected that trailing spaces are trimmed.", 2, iotaConfig.getMilestoneStartIndex());
+    }
+
+    /**
+     * Test if trailing spaces are correctly trimmed from boolean.
+     * @throws IOException when config file not found.
+     */
+    @Test
+    public void createFromFileTestnetWithBoolean() throws IOException {
+        File configFile = createTestnetConfigFile("true");
+        IotaConfig iotaConfig = ConfigFactory.createFromFile(configFile, true);
+        assertTrue("Expected that ZMQ is enabled.", iotaConfig.isZmqEnabled());
+    }
+
+    /**
+     * Try to create an {@link IotaConfig} from a not existing configFile.
+     * @throws IOException when config file not found.
+     */
+    @Test(expected = FileNotFoundException.class)
+    public void createFromFileTestnetWithFileNotFound() throws IOException {
+        File configFile = new File("doesNotExist.ini");
+        ConfigFactory.createFromFile(configFile, false);
+    }
+
+    private File createTestnetConfigFile(String testnet) throws IOException {
+        Properties properties = new Properties();
+        properties.setProperty("TESTNET", testnet);
+        properties.setProperty("ZMQ_ENABLED", " TRUE ");
```

Comment:
Note that the output in the file contains `\` for values with leading whitespace:
`ZMQ_ENABLED=\ TRUE`
Did you mean for that?